android/src/main/java/io/gdmp/GDMPCameraHelper.java

```diff
@@ -35,91 +34,110 @@ public class GDMPCameraHelper implements TextureFrameConsumer {
     private final Activity activity;
     private final EglManager eglManager;
     private final ExternalTextureConverter converter;
-    private final SingleThreadHandlerExecutor renderExecutor =
-            new SingleThreadHandlerExecutor("RenderThread", Process.THREAD_PRIORITY_DEFAULT);
-    private ProcessCameraProvider cameraProvider;
-    private Preview preview;
+    private final HandlerThread cameraThread;
+    private final Handler cameraHandler;
+    private CameraDevice cameraDevice;
+    private CameraCaptureSession captureSession;
+    private SurfaceTexture previewTexture;
 
     public GDMPCameraHelper(long nativeCallerPtr) {
         this.nativeCallerPtr = nativeCallerPtr;
         activity = GDMP.getSingleton().getGodot().getActivity();
-        // Get graph GL context by getting current context.
         eglManager = new EglManager(EGL14.eglGetCurrentContext());
         converter = new ExternalTextureConverter(eglManager.getContext());
         converter.setConsumer(this);
         converter.setFlipY(true);
         converter.setRotation(activity.getWindowManager().getDefaultDisplay().getRotation());
+        cameraThread = new HandlerThread("CameraThread", Process.THREAD_PRIORITY_DEFAULT);
+        cameraThread.start();
+        cameraHandler = new Handler(cameraThread.getLooper());
     }
 
     public void startCamera(int index, int width, int height) {
-        Size targetSize = new Size(width, height);
-        Executor mainThreadExecutor = ContextCompat.getMainExecutor(activity);
-        ListenableFuture<ProcessCameraProvider> cameraProviderFuture = ProcessCameraProvider.getInstance(activity);
-        cameraProviderFuture.addListener(() -> {
-            try {
-                cameraProvider = cameraProviderFuture.get();
-            } catch (Exception e) {
-                if (e instanceof InterruptedException) {
-                    Thread.currentThread().interrupt();
+        CameraManager cameraManager = (CameraManager) activity.getSystemService(Activity.CAMERA_SERVICE);
+        try {
+            String[] cameraIds = cameraManager.getCameraIdList();
+            Log.e(TAG, "Cameras found: " + Arrays.toString(cameraIds));
+            String cameraId;
+            if(index >= cameraIds.length) {
+                cameraId = cameraIds[0];
+            } else {
+                cameraId = cameraIds[index];
+            }
```

Comment:
Opening camera directly by index would create mismatch between currently defined camera facing enum and camera list returned by Camera2 (at least on my device, choosing front camera which is index 0, opened back camera).
If getCameraIdList always return front/back in fixed order, we can just swapping the front/back enum, but I do not know if that's the case, otherwise we would need additional check to ensure opened camera facing work as expected when camera facing enum is used.

Reply:
I think making it compatible with CameraFacing Enum is the best bet.
Camera2 does not guarantee which index is what camera so it has to be fetched by the `getCameraCharacteristics`  

You can try it now, index 0 is always the front camera and 1 is always the back one.

```java
           if (index == 0) {
                // Select front camera
                for (String id : cameraIds) {
                    CameraCharacteristics characteristics = cameraManager.getCameraCharacteristics(id);
                    Integer facing = characteristics.get(CameraCharacteristics.LENS_FACING);
                    if (facing != null && facing == CameraCharacteristics.LENS_FACING_FRONT) {
                        cameraId = id;
                        break;
                    }
                }
            } else if (index == 1) {
                // Select back camera
                for (String id : cameraIds) {
                    CameraCharacteristics characteristics = cameraManager.getCameraCharacteristics(id);
                    Integer facing = characteristics.get(CameraCharacteristics.LENS_FACING);
                    if (facing != null && facing == CameraCharacteristics.LENS_FACING_BACK) {
                        cameraId = id;
                        break;
                    }
                }
            } else {
                // Select other available camera
                if (index < cameraIds.length) {
                    cameraId = cameraIds[index];
                } else {
                    cameraId = cameraIds[0]; // Default to the first camera if index is out of bounds
                }
            }
 ```